activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -16,6 +16,47 @@ def self.find(version)
       V7_0 = Current
 
       class V6_1 < V7_0
+        class PostgreSQLCompat
+          def self.compatible_timestamp_type(type, connection)
```

Comment:
@pixeltrix @kamipo I've implemented the migration versioning here, tests are in `activerecord/test/cases/schema_dumper_test.rb`.

I think it's working correctly and tested for all scenarios:

- [x] Migration created in Rails 6.1, using default value for `datetime_type` -> no changes to schema.rb; `t.datetime` columns will continue to say `t.datetime`. The tests added for this pass without changes on `main`.
- [x] Migration created in Rails 6.1, with `datetime_type` changed to `:timestamptz` in an initializer -> if the migration said `t.datetime`, schema.rb will be changed to say `t.timestamp`. See comments + assertions [here](https://github.com/rails/rails/pull/41084/files#diff-f4545f25496b789b9cde6e484343a566d476078a22b3b770a80f3ec6b989ea64R605-R614) for why this is necessary.
- [x] Migration created in Rails 7+,  using default value for `datetime_type` -> `t.datetime` columns say `t.datetime` in schema.rb. `t.timestamp` columns say `t.datetime` in schema.rb. `t.timestamptz` columns say `t.timestamptz` in schema.rb.
- [x] Migration created in Rails 6.1, with `datetime_type` changed to `:timestamptz` -> `t.datetime` columns say `t.datetime` in schema.rb. `t.timestamp` columns say `t.timestamp` in schema.rb. `t.timestamptz` columns say `t.datetime` in schema.rb.